clap_derive/src/derives/clap.rs

```diff
@@ -11,486 +11,53 @@
-                        let #app_var = #arg_block;
-                        #app_var#from_attrs#version
-                    });
-                }
-            }
-        }
-    });
-
-    let app_methods = parent_attribute.top_level_methods();
-    let version = parent_attribute.version();
-    quote! {
-       pub fn augment_app<'b>(
-            app: ::clap::App<'b>
-        ) -> ::clap::App<'b> {
-            let app = app #app_methods;
-            #( #subcommands )*;
-            app #version
-        }
-    }
-}
-
-fn gen_from_subcommand(
-    name: &syn::Ident,
-    variants: &punctuated::Punctuated<syn::Variant, token::Comma>,
-    parent_attribute: &Attrs,
-) -> proc_macro2::TokenStream {
-    use syn::Fields::*;
-    let (flatten_variants, variants): (Vec<_>, Vec<_>) = variants
-        .iter()
-        .map(|variant| {
-            let attrs = Attrs::from_struct(
-                variant.span(),
-                &variant.attrs,
-                Name::Derived(variant.ident.clone()),
-                parent_attribute.casing(),
-                parent_attribute.env_casing(),
-            );
-            (variant, attrs)
-        })
-        .partition(|(_, attrs)| {
-            let kind = attrs.kind();
-            match &*kind {
-                Kind::Flatten => true,
-                _ => false,
-            }
-        });
-
-    let match_arms = variants.iter().map(|(variant, attrs)| {
-        let sub_name = attrs.cased_name();
-        let variant_name = &variant.ident;
-        let constructor_block = match variant.fields {
-            Named(ref fields) => from_argmatches::gen_constructor(&fields.named, &attrs),
-            Unit => quote!(),
-            Unnamed(ref fields) if fields.unnamed.len() == 1 => {
-                let ty = &fields.unnamed[0];
-                quote!( ( <#ty as ::clap::FromArgMatches>::from_argmatches(matches) ) )
-            }
-            Unnamed(..) => abort_call_site!("{}: tuple enums are not supported", variant.ident),
-        };
+pub fn derive_clap(input: &syn::DeriveInput) -> proc_macro2::TokenStream {
+    use syn::Data::*;
 
-        quote! {
-            (#sub_name, Some(matches)) => {
-                Some(#name :: #variant_name #constructor_block)
-            }
-        }
-    });
-    let child_subcommands = flatten_variants.iter().map(|(variant, _attrs)| {
-        let variant_name = &variant.ident;
-        match variant.fields {
-            Unnamed(ref fields) if fields.unnamed.len() == 1 => {
-                let ty = &fields.unnamed[0];
-                quote! {
-                    if let Some(res) = <#ty>::from_subcommand(other) {
-                        return Some(#name :: #variant_name (res));
-                    }
-                }
-            }
-            _ => abort!(
-                variant.span(),
-                "`flatten` is usable only with single-typed tuple variants"
-            ),
-        }
-    });
+    let struct_name = &input.ident;
```

Comment:
Let's rename this to `ident`

Reply:
I don't see much point, but if it make you feel any better...